test/sagemaker_tests/tensorflow/tensorflow2_training/integration/sagemaker/test_smtrcomp.py

```diff
@@ -405,17 +410,19 @@ def test_serving(self, sagemaker_session, ecr_image, framework_version, instance
                                framework_version=framework_version,
                                hyperparameters={
                                     TrainingCompilerConfig.HP_ENABLE_COMPILER : True,
+                                    'save-as-tf': True,
                                },
                                )
         estimator.fit(mnist_dataset, job_name=unique_name_from_base('test-TF-trcomp-serving'))
         _assert_model_exported_to_s3(estimator)
         captured = capsys.readouterr()
         _assert_training_compiler_invoked(captured)
-        predictor = estimator.deploy(initial_instance_count=1, instance_type=instance_type)
+        predictor = estimator.deploy(initial_instance_count=1, instance_type=instance_type,
+            image_uri=f'763104351884.dkr.ecr.{estimator.sagemaker_session.boto_region_name}.amazonaws.com/tensorflow-inference:2.10.0-gpu-py39-cu112-ubuntu20.04-sagemaker')
         predictor.delete_predictor()
 
 
-    @pytest.mark.xfail(reason="SM Neo does not currently support TF > 2.4")
+    @pytest.mark.xfail(reason="Neo does not support TF 2.11 yet. TF 2.11's saved models are not backwards compatible.")
```

Comment:
Would it be possible to put in an input guard on this test to ensure that it gets skipped for TF 2.11, instead of xfail-ing this test?

Reply:
I'd recommend keeping it as we'd know if the test is behaving as expected.